dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile

```diff
@@ -6,35 +6,30 @@ FROM docker.io/myoung34/github-runner:2.314.1-ubuntu-jammy
 # Basic packages.
 RUN apt update && \
     apt install -y \
-        wget python3.11 git unzip curl gnupg2 lsb-release vulkan-tools && \
+        wget python3.11 python3-pip git unzip curl gnupg2 lsb-release vulkan-tools && \
     update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
     update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3
 
-# Install the AMDVLK driver.
-WORKDIR /install-amdvlk
-RUN wget https://github.com/GPUOpen-Drivers/AMDVLK/releases/download/v-2023.Q3.1/amdvlk_2023.Q3.1_amd64.deb && \
-    dpkg -i amdvlk_2023.Q3.1_amd64.deb && \
-    rm -rf /install-amdvlk/*
+# CMake
+WORKDIR /install-cmake
+# Install the latest CMake version we support
+ARG CMAKE_VERSION="3.29.3"
+COPY build_tools/install_cmake.sh ./
+RUN ./install_cmake.sh "${CMAKE_VERSION}" && rm -rf /install-cmake
+
+# AMD GPU DRM & Vulkan
+WORKDIR /install-amdgpu
+ARG AMDGPU_VERSION=6.1
+COPY build_tools/install_amdgpu.sh ./
+RUN ./install_amdgpu.sh "${AMDGPU_VERSION}" && rm -rf /install-amdgpu
 WORKDIR /
-RUN rmdir /install-amdvlk
 
-# Install the ROCM driver.
-# In order to use this in the container, you must pass /dev/dri and /dev/kfd in as:
-#   docker run --device=/dev/kfd --device=/dev/dri <docker ID> rocminfo
-ARG ROCM_VERSION=5.6
-WORKDIR /install-rocm
-RUN mkdir --parents --mode=0755 /etc/apt/keyrings && wget https://repo.radeon.com/rocm/rocm.gpg.key -O - | \
-    gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null && \
-    echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/amdgpu/${ROCM_VERSION}/ubuntu jammy main" | \
-    tee /etc/apt/sources.list.d/amdgpu.list && \
-    echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${ROCM_VERSION} jammy main" | \
-    tee /etc/apt/sources.list.d/rocm.list && \
-    echo -e 'Package: *\nPin: release o=repo.radeon.com\nPin-Priority: 600' | tee /etc/apt/preferences.d/rocm-pin-600 && \
-    apt update && DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends ca-certificates rocm-hip-libraries && \
-    echo "/opt/rocm/lib\n/opt/rocm/hip/lib" | tee /etc/ld.so.conf.d/24-rocm.conf && ldconfig && \
-    rm -rf /install-rocm/*
+# TheRock (ROCm)
+WORKDIR /install-the-rock
+COPY build_tools/install_the_rock.sh ./
+RUN ./install_the_rock.sh \
+  && rm -rf /install-the-rock
 WORKDIR /
-RUN rmdir /install-rocm/
 
```

Comment:
I am curious why we don't do
```
RUN apt clean && rm -rf /var/lib/apt/lists/*
```
to remove the apt artifacts in the github image, but we do it in the ubuntu stock image? Is this intentional?

Reply:
I'd have to look at each, but generally adding that as a separate run step doesn't help anything. Each RUN gets exported as a layer, so doing it separately just means that you have all of that in a previous layer and then you make a final later that records the deletions (but doesn't actually save any image space).

The right way to do package stuff is to script it all in one script then, in I've RUN invoke the script and && any cleanup tasks. 

Not all docker images were done right.

Reply:
I added the cleanup to the github image as well.

Reply:
The problem is that multiple steps may use apt and then if you cleanup after each step then the next step would repeat stuff and slow down the image creation.